.github/workflows/check_build.yml

```
@@ -8,33 +8,24 @@ on: [push, pull_request]
 
 jobs:
   build:
-    strategy:
-      matrix:
-        # Use these Java versions
-        java: [
-          21,  # Minimum
-        ]
-        os: [ubuntu-22.04]
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-22.04
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
       - name: Validate Gradle wrapper
         uses: gradle/wrapper-validation-action@v1
-      - name: Setup JDK ${{ matrix.java }}
+      - name: Setup JDK 21
         uses: actions/setup-java@v3
         with:
           distribution: zulu
-          java-version: ${{ matrix.java }}
+          java-version: 21
       - name: Make Gradle wrapper executable
-        if: ${{ runner.os != 'Windows' }}
         run: chmod +x ./gradlew
       - name: Build
         run: ./gradlew build --stacktrace --parallel
       - name: Capture build artifacts
-        if: ${{ runner.os == 'Linux' && matrix.java == '17' }} # Only upload artifacts built from LTS java on one OS
         uses: actions/upload-artifact@v3
```

```suggestion
        uses: actions/upload-artifact@v4
```
You should also update `actions/upload-artifact` to version 4, [otherwise it'll stop working again soon](https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/).